disnake/client.py

```diff
@@ -194,6 +199,26 @@ def __repr__(self):
         )
 
 
+class GatewayParams(NamedTuple):
```

Comment:
a frozen dataclass would make more sense imo, since each argument can be required as a keyword argument.

Reply:
`kw_only` is a 3.10 feature, sadly

Reply:
if we wanted to add a direct dependency (which is a transitive dependency already because aiohttp relies on it) we could use attrs. That's out of the scope of this tho.